### PR TITLE
Cleanup `MLflowCallback` tests

### PR DESCRIPTION
**Pull request by xadrianzetx**
## Motivation

A bit overdue cleanup of `MLflowCallback` tests as discussed in https://github.com/optuna/optuna/pull/2912#issuecomment-915462924 and https://github.com/optuna/optuna/pull/2912#discussion_r720187136.

## Description of the changes

* Reduced test duplication
* Fixed `tracking_uri` naming
* Refactored `test_log_*` to use only public APIs
* simplified `test_study_name`

### CODE REVIEW

**Comment by himkt** [comment]
Thank you for the PR!

@keisuke-umezawa @nzw0301 Could you please take a look?
Also please comment @toshihikoyanase as the former reviewer of related PRs.

**Comment by nzw0301** [comment]
Due to my main job, I'll be busy this week, so it might take time to review this PR 🙇🏻

File: tests/integration_tests/test_mlflow.py [hunk 1]
```diff
@@ -52,47 +56,22 @@ def _objective_func_long_user_attr(trial: optuna.trial.Trial) -> float:
...
-    )
-    assert (
-        first_run_dict["data"]["tags"]["y_distribution"]
-        == "LogUniformDistribution(high=30.0, low=20.0)"
-    )
-    assert (
-        first_run_dict["data"]["tags"]["z_distribution"]
-        == "CategoricalDistribution(choices=(-1.0, 1.0))"
-    )
-    assert first_run_dict["data"]["tags"]["my_user_attr"] == "my_user_attr_value"
```
Comment (HideakiImamura) [inline tests/integration_tests/test_mlflow.py:95]: Could you explain why you removed these lines?

File: tests/integration_tests/test_mlflow.py [hunk 2]
```diff
@@ -425,40 +423,28 @@ def _objective_func(trial: optuna.trial.Trial) -> float:
...
-    study = optuna.create_study(study_name=study_name)
-
-    mlflc._initialize_experiment(study)
-
-    mlfl_client = MlflowClient(tracking_file_name)
-    experiments = mlfl_client.list_experiments()
-    assert len(experiments) == 1
-
-    experiment = experiments[0]
-    assert experiment.name == study_name
```
Comment (HideakiImamura) [inline tests/integration_tests/test_mlflow.py:443]: Ditto.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have a naive question.

File: tests/integration_tests/test_mlflow.py (hunk 2 above)
Comment (xadrianzetx) [reply tests/integration_tests/test_mlflow.py:443]: `test_study_name` covers this already, so I removed this duplication.

Comment (nzw0301) [reply tests/integration_tests/test_mlflow.py:443]: I think this test function aims to verify `mlflc._initialize_experiment(study)`; if we remove `mlflc._initialize_experiment` in this test, we need to rename the test function name or remove this test. So we might need the original test case. 

I suppose `test_log_metric`, `test_log_metric_none`, and `test_log_params` are also same; they aim to verify a primate method of the callaback.

Comment (xadrianzetx) [reply tests/integration_tests/test_mlflow.py:443]: I think the same can be archived by testing only using public APIs, regardless of callback implementation details. We test for specific callback behavior (is experiment created with correct name?) which in this case is covered by another test. The test will stay the same even if `mlflc._initialize_experiment` is removed. Same goes for `log_*`

File: tests/integration_tests/test_mlflow.py (hunk 1 above)
Comment (HideakiImamura) [reply tests/integration_tests/test_mlflow.py:95]: Thanks for the explanation. I found `test_tag_trial_user_attrs` tests the user attrs. Looks good.

File: tests/integration_tests/test_mlflow.py (hunk 2 above)
Comment (HideakiImamura) [reply tests/integration_tests/test_mlflow.py:443]: Sounds reasonable. Let me resolve this conversation.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the explanations! LGTM!

**Review by nzw0301 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**